docs/Account/methods.md

```diff
@@ -0,0 +1,284 @@
+
+      ```ts
+      type BigNumberish = BigNumber | Bytes | bigint | string | number;
+
+      type Overrides = {
+        callGasLimit?: BigNumberish;
+        verificationGasLimit?: BigNumberish;
+        preVerificationGas?: BigNumberish;
+        maxFeePerGas?: BigNumberish;
+        maxPriorityFeePerGas?: BigNumberish;
+        paymasterData?: string;
+        signature?: string; // todo - add the definition
+      }
+      ```
+
+  2. skipBundlerGasEstimation (`boolean`): This boolean param can be set to false to skip some Gas Estimations on the bundler which can help with reducing latency on requests.
+
+  3. params (`ModuleInfo`): One can use this param to pass session validation module parameters. Refer to the tutorial to learn more about the session keys.
+
+      ```ts
+      type ModuleInfo = {
+        sessionID?: string;
+        sessionSigner?: Signer;
+        sessionValidationModule?: string;
+        additionalSessionData?: string;
+        batchSessionParams?: SessionParams[];
+      }
+      ```
+  4. nonceOptions(`NonceOptions`) : this can be used to execute multiple transactions in parallel. One can implement their own logic for nonce management.
+
+      ```ts
+      type NonceOptions = {
+        nonceKey?: number;
+        nonceOverride?: number;
+      };
+      // nonceOptions usage
+      let i = 0;
+      const userOp = await smartAccount.buildUserOp([tx1], {
+        nonceOptions: { nonceKey: i++ },
+      });
+      ```
+      Nonce can be initialised at any arbitrary number and incremented as one builds transactions. The nonceKey will create a batch or space in which the nonce can safely increment without colliding with other transactions. The nonceOverride will directly override the nonce and should only be if you know the order in which you are sending the userOps.
+
+    
+    
+  5. forceEncodeForBatch (`boolean`): 
+
+  6. paymasterServiceData (`SponsorUserOperationDto`): The `paymasterServiceData` includes details about the kind of sponsorship and payment token. It contains information about the paymaster service, which is used to calculate the `paymasterAndData` field in the user operation.
+
+      ```ts
+      type SponsorUserOperationDto = {
+        mode: PaymasterMode;
+        calculateGasLimits?: boolean; // Add info
+        expiryDuration?: number;
+        webhookData?: {
+            [key: string]: any;
+        };
+        smartAccountInfo?: SmartAccountData;
+        feeTokenAddress?: string;
+      }
+      ```
+      It also contains optional fields such as `webhookData` and `smartAccountInfo`.
+  
+
+**Returns**
+- partialUserOp (`Promise<Partial<UserOperation>>`): A Promise resolving to `partialUserOp` which can be further sent for execution⁠.
+
+### senduserOp( )
+
+This method is used to submit a User Operation object to the User Operation pool of the client. It signs the UserOperation using an externally owned account (EOA) key and submits it to the bundler for on-chain processing. 
```

Comment:
not always using an externally owned account (EOA) key

Reply:
signs the useroperation with the help of instance of active validation module.

dev -> smartAccount.sendUserop -> smartAccount.signUserOp -> activeValidationModule.signUserOp

Reply:
could you explain more on signing with the instance of active VM

Reply:
it happens internally. 
check this 
https://github.com/bcnmy/biconomy-client-sdk/blob/main/packages/account/src/BiconomySmartAccountV2.ts#L355

https://github.com/bcnmy/biconomy-client-sdk/blob/main/packages/account/src/BiconomySmartAccountV2.ts#L372

Reply:
my suggestion from now on keep a copy of client sdk repo open and try to trace API methods etc when you don't understand something. we can also do a KT session on sdk walkthrough